src/2d/errf1.f

```diff
@@ -63,26 +63,34 @@ subroutine errf1(rctfine,nvar,rctcrse,mptr,mi2tot,mj2tot,
       ifine = nghost+1
 c
       do 30  i  = nghost+1, mi2tot-nghost
-          rflag = goodpt
-          xofi  = xleft + (dble(ifine) - .5d0)*hx
-          term1 = rctfine(1,ifine,jfine)
-          term2 = rctfine(1,ifine+1,jfine)
-          term3 = rctfine(1,ifine+1,jfine+1)
-          term4 = rctfine(1,ifine,jfine+1)
-c         # divide by (aval*order) for relative error
-          aval  = (term1+term2+term3+term4)/4.d0
-          est   =  dabs((aval-rctcrse(1,i,j))/ order)
-          if (est .gt. errmax) errmax = est
-          err2 = err2 + est*est
-c         write(outunit,102) i,j,est,rctcrse(1,i,j)
- 102      format(' i,j,est ',2i5,2e15.7)
-c          write(outunit,104) term1,term2,term3,term4
- 104      format('   ',4e15.7)
-c         rctcrse(2,i,j) = est
-c
-          if (est .ge. tol) then
-             rflag  = badpt
-          endif 
+          rflag = UNSET
+c Only check errors if flag hasn't been set yet.
+c If flag == DONTFLAG then refinement is forbidden by a region,
+c if flag == DOFLAG checking is not needed
+          if(rctflg(ifine,jfine) == UNSET
+     .         .or. rctflg(ifine+1,jfine) == UNSET
+     .         .or. rctflg(ifine,jfine+1) == UNSET
+     .         .or. rctflg(ifine+1,jfine+1) == UNSET) then
+              xofi  = xleft + (dble(ifine) - .5d0)*hx
+              term1 = rctfine(1,ifine,jfine)
+              term2 = rctfine(1,ifine+1,jfine)
+              term3 = rctfine(1,ifine+1,jfine+1)
+              term4 = rctfine(1,ifine,jfine+1)
+c             # divide by (aval*order) for relative error
+              aval  = (term1+term2+term3+term4)/4.d0
+              est   =  dabs((aval-rctcrse(1,i,j))/ order)
```

Comment:
I'm confused about what this routine is doing (also the old version in master). At this line it seems like `rctcrse` and `rctfine` are solution values, and that is what is passed in as `alloc(locbig)` and `valbgc` from `errest`, I believe.

But in line 88 below, `rctcrse(1,i,j) = rflag` so it is being used as a flag?  

Maybe this is correct and `rctcrse` is just being temporarily used to hold the flag on the coarse grid that gets passed down to the fine grid `rctflg` in  the next loop.  If so this seems confusing. Would it be better to define a new local variable `rctflgc`, for example?  At least we need a comment or two.

Maybe @mjberger can comment.